library/std/src/sys/pal/uefi/helpers.rs

```diff
@@ -330,6 +389,99 @@ impl<'a> crate::fmt::Debug for BorrowedDevicePath<'a> {
     }
 }
 
+pub(crate) struct DevicePathIterator<'a>(Option<DevicePathNode<'a>>);
```

Comment:
Can you move the addition of the iterator to a separate PR?

Reply:
Sure. I have limited bandwidth rn, so will do it in a few weeks.

Reply:
I have created the PR: https://github.com/rust-lang/rust/pull/137424